src/typechecker.erl

```diff
@@ -3438,16 +3438,63 @@ refinable(_) ->
 no_guards({clause, _, _, Guards, _}) ->
     Guards == [].
 
+
+-spec check_guard_call(atom(), list()) -> map().
+check_guard_call(is_atom, [{var, _, Var}]) -> #{Var => type(atom)};
+check_guard_call(is_binary, [{var, _, Var}]) -> #{Var => type(binary)};
+check_guard_call(is_bitstring, [{var, _, Var}]) -> #{Var => type(bitstring)};
+check_guard_call(is_boolean, [{var, _, Var}]) -> #{Var => type(boolean)};
+check_guard_call(is_float, [{var, _, Var}]) -> #{Var => type(float)};
+check_guard_call(is_function, [{var, _, Var}]) -> #{Var => type('fun')};
+check_guard_call(is_function, [{var, _, Var}, _]) -> #{Var => type('fun')};
```

Comment:
We can actually return a more precise type here if the second argument to `is_function` is a literal number. We would then know the arity of the function. For example, if we encountered `is_function(F,2)` we can give `F` the type  `fun(any(), any()) -> any()`.

It's fine if you don't want to make this improvement in this PR, but I think it's a natural thing to include.

Reply:
Nice. Yea I wasn't sure how to do this so I didn't implement it. Will do.

Reply:
Done with caveats. Added a known_issue because I can't seem to get tests to fail but I emit the proper types.